client/layout/sidebar-v2/navigator/navigator-menu-item.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+	__experimentalNavigatorButton as NavigatorButton,
+	__experimentalItem as Item,
+	__experimentalHStack as HStack,
+	FlexBlock,
+} from '@wordpress/components';
+import { Icon, chevronRightSmall } from '@wordpress/icons';
+import classnames from 'classnames';
+
+import './style.scss';
+
+const ICON_SIZE = 24;
+
+interface Props {
+	icon: JSX.Element;
+	path: string;
+	link: string;
+	title: string;
+	onClickMenuItem: ( path: string ) => void;
+	withChevron?: boolean;
+}
+
+export const SidebarNavigatorMenuItem = ( {
+	icon,
+	path,
+	link,
+	title,
+	onClickMenuItem,
+	withChevron,
+}: Props ) => {
+	const pathname = window.location.pathname;
+
+	const SidebarItem = ( { children }: { children?: JSX.Element } ) => {
+		return (
+			<Item
+				className={ classnames( 'sidebar-v2__menu-item', {
+					'is-active': pathname === link,
+				} ) }
+				onClick={ () => onClickMenuItem( link ) }
+			>
+				<HStack justify="flex-start">
+					{ icon && <Icon style={ { fill: 'currentcolor' } } icon={ icon } size={ ICON_SIZE } /> }
+					<FlexBlock>{ children }</FlexBlock>
+					{ withChevron && <Icon icon={ chevronRightSmall } size={ ICON_SIZE } /> }
```

Comment:
Can we also add an optional right icon or `isExternalLink` to show an external icon like the use case below?

<img width="495" alt="Screen Shot 2023-10-17 at 3 13 58 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/cc85775e-ec15-49eb-90a1-06fc8e762555">

Reply:
Thanks for the suggestion. Added it here: https://github.com/Automattic/wp-calypso/pull/83073/commits/9b5d6043e59f010e9a2f9bdfcc54da0ed61ef889

Reply:
Thanks for addressing this use case. 🙇